### DISCUSSION TITLE
Omit disconnected fragment substructures

### DISCUSSION THREAD

**Discussion by ConstantinWaquet** [rdkit/rdkit#4734] [Q&A] [answered]
Does anyone know how to get FindMCS to find contiguous maximum common substructures only?

At the moment the following code:

```
#import the relevant modules
from rdkit import Chem
from rdkit.Chem import rdFMCS

#get the molecules as RDKit mol objects
Smile1 = "Fc1ccc(-c2ccc3c(c2)[nH]c2ccncc23)cn1"
Smile2 = "C[C@]12CC[C@@H]3c4ccc(O)cc4CC[C@H]3[C@@H]1C[C@@H]([18F])[C@@H]2O"
RDMols = [Chem.MolFromSmiles(Smile1), Chem.MolFromSmiles(Smile2)]

#use the rdFMCS module to find the  substructure
MCS = rdFMCS.FindMCS(RDMols)

#view this in a .ipynb to see the output (pictured below)
Chem.MolFromSmarts(MCS.smartsString)
```

gives the following common substructure. How do I only get substructures which are not separate disconnected fragments?

![df40a7bb-dfe6-4bef-ae83-2da87cd38618](https://user-images.githubusercontent.com/92093750/143432756-80ac0a33-c6a2-4670-90cf-35069026afa8.png)

**Comment by ptosco** [comment, marked as answer]
@ConstantinWaquet They are not disconnected, the dotted grey lines indicate 'any' bonds. RDKit MCS are always constituted by a single connected fragment.
You can visualize the connected substructure matches on your molecules through highlighting  as follows:

```
mcs_query = Chem.MolFromSmarts(MCS.smartsString)
atom_matches = [m.GetSubstructMatch(mcs_query) for m in RDMols]
inverse_atom_matches = [{j: i for i, j in enumerate(match)} for match in atom_matches]
bond_matches = [[b.GetIdx() for b in m.GetBonds()
                 if b.GetBeginAtomIdx() in atom_matches[i] and b.GetEndAtomIdx() in atom_matches[i] and
                 mcs_query.GetBondBetweenAtoms(inverse_atom_matches[i][b.GetBeginAtomIdx()],
                                               inverse_atom_matches[i][b.GetEndAtomIdx()])] for i, m in enumerate(RDMols)]
MolsToGridImage(RDMols, highlightAtomLists=atom_matches, highlightBondLists=bond_matches)
```
![image](https://user-images.githubusercontent.com/5244385/143437417-a1e84049-19c0-4f3a-827e-049ea18ba6a2.png)

**Reply by ConstantinWaquet** [reply]
That makes a lot of sense.. Thank you very much!
